core-bundle/src/Resources/public/autofocus.js

```diff
@@ -0,0 +1,37 @@
+/*!
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+window.addEventListener('DOMContentLoaded', function () {
+    const edit = document.querySelector('#main .tl_formbody_edit');
+    if (!edit) return;
+
+    // Copy from MooTools
+    function isHidden (el) {
+        const w = el.offsetWidth, h = el.offsetHeight,
+            force = /^tr$/i.test(el.tagName);
+        return (w===0 && h===0 && !force) ? true : (w!==0 && h!==0 && !force) ? false : el.style.display === 'none';
+    }
+
+    const inputs = edit.querySelectorAll('input, textarea');
+
+    for (let i = 0; i < inputs.length; i++) {
+        const input = inputs[i];
+
+        if (
+            !input.getAttribute('disabled')
+            && !input.getAttribute('readonly')
+            && !isHidden(input)
```

Comment:
```suggestion
            && input.offsetWidth
            && input.offsetHeight
```

Reply:
That would not check for the `display: none` property (according to the Mootols code)?

Reply:
Elements with `display: none` have `offsetWidth` and `offsetHeight` of `0`.

Reply:
Changed in b4b4b733ab915ced966e20ea3f073d28bc27d52f